libpressio-sys/build.rs

```diff
@@ -7,12 +7,38 @@ fn main() {
     println!("cargo::rerun-if-changed=build.rs");
     println!("cargo::rerun-if-changed=wrapper.h");
     println!("cargo::rerun-if-changed=libpressio");
+    println!("cargo::rerun-if-changed=std_compat");
 
+    // ---------------------------------------------------------
+    // Configure std_compat, the compiler portability layer
+    // ---------------------------------------------------------
+    let mut stdcompat_config = cmake::Config::new("std_compat");
+    //prefer static libraries for RUST
+    stdcompat_config.define("BUILD_SHARED_LIBS", "OFF");
+    //disable testing to avoid google-test dependency
+    stdcompat_config.define("BUILD_TESTING", "OFF");
+    // require a C++17 compiler (e.g. gcc 12 or later) for now
+    // this includes Ubuntu 24.04 and later, Fedora, Nyx, et al 
+    // https://robertu94.github.io/guides/dependencies
+    stdcompat_config.define("STDCOMPAT_CXX_VERSION", "17");
+    stdcompat_config.define("STDCOMPAT_CXX_UNSTABLE", "ON");
+    stdcompat_config.define("STD_COMPAT_BOOST_REQUIRED", "OFF");
+    let stdcompat_out = stdcompat_config.build();
+    println!("cargo:rustc-link-search=native={}", stdcompat_out.display());
+
+    // ---------------------------------------------------------
+    // Configure libpressio
+    // ---------------------------------------------------------
     let mut config = cmake::Config::new("libpressio");
-    let out = config.build();
+    config.define("BUILD_SHARED_LIBS", "OFF");
+    config.define("BUILD_TESTING", "OFF");
+    config.define("LIBPRESSIO_HAS_OPENMP", "ON");
```

Comment:
This should be configurable (at least I've never tried building with OpenMP on WASM so far):

- [ ] give libpressio-sys an openmp feature, e.g. https://github.com/juntyr/tthresh-rs/blob/454cdba7fe763bd4b072208f635e82dcda48ae72/Cargo.toml#L71
- [ ] toggle the "ON" or "OFF" with a `cfg!(feature = "openmp")`
- [ ] give libpressio an openmp feature that forwards to libpressio-sys: https://github.com/juntyr/tthresh-rs/blob/454cdba7fe763bd4b072208f635e82dcda48ae72/Cargo.toml#L71